ipfs-cluster-service/migrate.go

```diff
@@ -0,0 +1,91 @@
+package main
+
+import (
+	"bufio"
+
+	"fmt"
+	"os"
+
+	"github.com/urfave/cli"
+
+	peer "github.com/libp2p/go-libp2p-peer"
+	
+	ipfscluster "github.com/ipfs/ipfs-cluster"
+	"github.com/ipfs/ipfs-cluster/state/mapstate"
+	"github.com/ipfs/ipfs-cluster/consensus/raft"
+)
+
+
+func upgrade(c *cli.Context) error {
+	if c.NArg() < 1 || c.NArg() > 2 {
```

Comment:
do the argument checking on the Action() and then call upgrade() with the right values